src/components/MAPI/releases/ClusterDetailWidgetRelease.tsx

```diff
@@ -134,61 +159,150 @@ const ClusterDetailWidgetRelease: React.FC<IClusterDetailWidgetReleaseProps> = (
-        value={releaseVersion}
-        replaceEmptyValue={false}
-      >
-        {(value) => (
-          <Keyboard onSpace={handleVersionClick}>
-            <StyledLink
-              href='#'
-              aria-label={`Cluster release version ${value}`}
-              onClick={handleVersionClick}
-            >
-              <Text>
-                <i
-                  className='fa fa-version-tag'
-                  role='presentation'
-                  aria-hidden='true'
-                />
-              </Text>{' '}
-              <VersionLabel>{value || <NotAvailable />}</VersionLabel>
-            </StyledLink>
-          </Keyboard>
-        )}
-      </ClusterDetailWidgetOptionalValue>
-      <StyledDot />
-      <ClusterDetailWidgetOptionalValue
-        value={k8sVersion}
-        replaceEmptyValue={false}
-      >
-        {(value) => (
-          <KubernetesVersionLabel
-            hidePatchVersion={false}
-            version={value as string}
+    <ClusterDetailWidget title='Release' inline={true} {...props}>
+      <Box direction='row' gap='xsmall' wrap={true} align='center'>
+        <ClusterDetailWidgetOptionalValue
+          value={releaseVersion}
+          replaceEmptyValue={false}
+        >
+          {(value) => (
+            <Keyboard onSpace={handleVersionClick}>
+              <StyledLink
+                href='#'
+                aria-label={`Cluster release version ${value}`}
+                onClick={handleVersionClick}
+              >
+                <Text>
+                  <i
+                    className='fa fa-version-tag'
+                    role='presentation'
+                    aria-hidden='true'
+                  />
+                </Text>{' '}
+                <VersionLabel>{value || <NotAvailable />}</VersionLabel>
+              </StyledLink>
+            </Keyboard>
+          )}
+        </ClusterDetailWidgetOptionalValue>
+        <StyledDot />
+        <ClusterDetailWidgetOptionalValue
+          value={k8sVersion}
+          replaceEmptyValue={false}
+        >
+          {(value) => (
+            <KubernetesVersionLabel
+              hidePatchVersion={false}
+              version={value as string}
+            />
+          )}
+        </ClusterDetailWidgetOptionalValue>
+
+        {cluster && (
+          <ClusterDetailStatus
+            isDeleting={isDeleting}
+            isUpgrading={isUpgrading}
+            isUpgradable={isUpgradable}
+            margin={{ left: 'small' }}
           />
         )}
-      </ClusterDetailWidgetOptionalValue>
-
-      {cluster && (
-        <ClusterDetailStatus
-          isDeleting={isDeleting}
-          isUpgrading={isUpgrading}
-          isUpgradable={isUpgradable}
-          margin={{ left: 'small' }}
-        />
+      </Box>
+
+      {!isCreating && !isUpgrading && isUpgradable && (
+        <Box margin={{ top: 'small' }}>
+          <Button onClick={handleUpgradeButtonClick}>Upgrade cluster…</Button>
```

Comment:
The upgrade button is not displayed if the cluster is still creating or upgrading. In these cases, an upgrade would error out on the admission controller side.